roles/backend_setup/tasks/cache_setup.yml

```diff
@@ -2,34 +2,41 @@
 # This playbook sets up lvm cache. Primarily written for the hyperconverged
 # setup.
 
+# If the ssd_disk is provided then ssd_vgname has to be provided
+- fail:
+    msg: "If gluster_infra_ssd_disk is provided gluster_infra_ssd_vgname is needed"
+  when: gluster_infra_ssd_disk is defined and
+        gluster_infra_ssd_vgname is not defined
+
+
 # Create PV for setting up cache with dataalignment of 256K for SSDs.
 # Extend the existing volume group with the SSD
 # To extend the vg, pvs should list the earlier pv as well.
-- name: Extend volume group - "{{ gluster_infra_vg_name }}"
+- name: Extend volume group
   lvg:
      state: present
-     vg: "{{ gluster_infra_vg_name }}"
+     vg: "{{ gluster_infra_ssd_vgname }}"
      pvs: "{{ gluster_infra_pvs }},{{ gluster_infra_ssd_disk }}"
```

Comment:
Doesn't seem possible to cache a vg created using the multiple vgs option. Which means that only 1 VG can provisioned with a cache at a time.